src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -0,0 +1,179 @@
+		ValidationResult  result = config.validate();
+
+		if(result.getErrors().size() > 0){
+			throw new PermanentEventNotificationException("Please verify your webhook URL");
+		}
+
+		try {
+			SlackMessage slackMessage = createSlackMessage(ctx, config);
+			slackClient.send(slackMessage,config.webhookUrl());
+		} catch (Exception e) {
+			final Notification systemNotification = notificationService.buildNow()
+					.addNode(nodeId.toString())
+					.addType(Notification.Type.GENERIC)
+					.addSeverity(Notification.Severity.NORMAL)
+					.addDetail("exception", e.toString());
+
+			notificationService.publishIfFirst(systemNotification);
+			throw new TemporaryEventNotificationException("Slack notification is triggered, but sending failed. " + e.getMessage(), e);
+
+		}
+
+	}
+
+	/**
+	 *
+	 * @param ctx
+	 * @param config
+	 * @return
+	 * @throws PermanentEventNotificationException - throws this exception when the custom message template is invalid
+	 */
+	SlackMessage createSlackMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) throws PermanentEventNotificationException {
+		//Note: Link names if notify channel or else the channel tag will be plain text.
+		boolean linkNames = config.linkNames() || config.notifyChannel();
+		String message = buildDefaultMessage(ctx, config);
+
+		String customMessage = null;
+		String template = config.customMessage();
+		if (!isNullOrEmpty(template)) {
+			customMessage = buildCustomMessage(ctx, config, template);
+		}
+
+		return new SlackMessage(
+				config.color(),
+				config.iconEmoji(),
+				config.iconUrl(),
+				config.userName(),
+				config.channel(),
+				linkNames,
+				message,
+				customMessage
+				);
+	}
+
+	String buildDefaultMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) {
+		String title = buildMessageTitle(ctx);
+
+		// Build custom message
+		String audience = config.notifyChannel() ? "@channel " : "";
+		String description = ctx.eventDefinition().map(EventDefinitionDto::description).orElse("");
+		return String.format(Locale.ROOT,"%s*Alert %s* triggered:\n> %s \n", audience, title, description);
+	}
+
+	private String buildMessageTitle(EventNotificationContext ctx) {
+		String eventDefinitionName = ctx.eventDefinition().map(EventDefinitionDto::title).orElse("Unnamed");
+		return "_" + eventDefinitionName + "_";
+	}
+
+	String buildCustomMessage(EventNotificationContext ctx, SlackEventNotificationConfig config, String template) throws PermanentEventNotificationException {
+		List<MessageSummary> backlog = getAlarmBacklog(ctx);
+		Map<String, Object> model = getCustomMessageModel(ctx, config, backlog);
+		try {
+			LOG.debug("template = {} model = {}" ,template, model);
+			return templateEngine.transform(template, model);
+		} catch (Exception e) {
+			LOG.error("Exception during templating", e);
+			throw new PermanentEventNotificationException(e.toString(),e.getCause());
```

Comment:
this needs some thought.

Reply:
the test `buildCustomMessage_with_invalidTemplate` does throw `PermanentEventNotificationException` . @waab76 is the test incorrect ?

Reply:
Your test only covers the behavior of the `buildCustomMessage` method.  What I'm saying is that `buildCustomMessage` throws a `PermanentEventNotificationException` which then gets caught by the try/catch block in `execute` and replaced with a `TemporaryEventNotificationException`.  If you believe it is right to throw a `PermanentEventNotificationException` here, then your try/catch block in `execute` should specifically catch `PermanentEventNotificationException` and re-throw it rather than wrapping it in a `TemporaryEventNotificationException`.